cpp/tests/replace/replace_tests.cu

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tests/utilities/legacy/cudf_test_fixtures.h>
+#include <tests/utilities/legacy/cudf_test_utils.cuh>
+#include <tests/utilities/legacy/column_wrapper.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/replace.hpp>
+
+#include <cudf/cudf.h>
+
+#include <gtest/gtest.h>
+
+// This is the main test feature
+template <class T>
+struct ReplaceTest : public GdfTest
+{
+
+  ReplaceTest()
+  {    
+  }
+
+  ~ReplaceTest()
+  {
+  }
+};
+
+// ugly.  only here to get around strict-aliasing compiler warnings
+uint32_t hard_cast_float(float val)
+{
+   void *v_val = &val;
+   return *((uint32_t*)v_val);
```

Comment:
```suggestion
   return *reinterpret_cast<uint32_t*>(&val);
```

Reply:
Fixed in ff7f197e70fcccca7539a90eec5fade7f986e36f.   I'm using expect_equal_buffers now so this whole thing goes away.